app-backend/db/src/main/scala/AnnotationProjectDao.scala

```diff
@@ -89,55 +90,56 @@ object AnnotationProjectDao
       page: PageRequest,
       params: AnnotationProjectQueryParameters,
       user: User
-  ): ConnectionIO[PaginatedResponse[AnnotationProject.WithRelated]] =
-    authQuery(
+  ): ConnectionIO[
+    PaginatedResponse[AnnotationProject.WithRelatedAndSummary]
+  ] =
+    (authQuery(
       user,
       ObjectType.AnnotationProject,
       params.ownershipTypeParams.ownershipType,
       params.groupQueryParameters.groupType,
       params.groupQueryParameters.groupId
     ).filter(params)
-      .page(page)
-      .flatMap(toWithRelated)
-
-  def toWithRelated(
-      projectsPage: PaginatedResponse[AnnotationProject]
-  ): ConnectionIO[PaginatedResponse[AnnotationProject.WithRelated]] =
-    projectsPage.results.toList.toNel match {
-      case Some(projects) =>
-        projects traverse { project =>
-          for {
-            tileLayers <- TileLayerDao.listByProjectId(project.id)
-            labelClassGroups <- AnnotationLabelClassGroupDao.listByProjectId(
-              project.id
-            )
-            labelClassGroupsWithClasses <- labelClassGroups traverse {
-              labelClassGroup =>
-                AnnotationLabelClassDao.listAnnotationLabelClassByGroupId(
-                  labelClassGroup.id
-                ) map { cls =>
-                  labelClassGroup.withLabelClasses(cls)
+      .list(page.offset, page.limit, Page.createOrderClause(page))
+      .map { projectList =>
+        projectList traverse {
+          project =>
+            getWithRelatedAndSummaryById(project.id) map {
+              case Some(project) =>
+                val taskStatusList =
+                  params.projectFilterParams.taskStatusesInclude.toList
+                if (taskStatusList.size > 0) {
```

Comment:
the big if-else with nested bool matching and option matching makes me feel like there's a cleaner way to do this -- I can't comment on the whole range right now, but after you answer that question I'd like to see if we can:

- avoid the double `flatten` at the end
- avoid the `bool match { case true => ... }`
- and not check the list size

Reply:
Tests for added functionality were too good and revealed that Scala-side filtering violated expectations about pages -- so new work is to rework how the group summaries are calculated to make the status filters possible in the initial DB request for annotation projects. Upside is this cleanup is no longer necessary, since instead it will be lit on :fire: